src/sentry/api/helpers/group_index/delete.py

```diff
@@ -45,13 +46,18 @@ def delete_group_list(
     eventstream_state = eventstream.backend.start_delete_groups(project.id, group_ids)
     transaction_id = uuid4().hex
 
+    all_group_hashes = GroupHash.objects.filter(project_id=project.id, group__id__in=group_ids)
+
+    # Tell seer to delete grouping records with these group hashes
+    if features.has("projects:similarity-embeddings-delete-by-hash", project):
+        # TODO (jangjodi): once we store seer grouping info in GroupHash, we should filter by that here
+        delete_seer_grouping_records_by_hash.apply_async(
+            args=[project.id, [group_hash.hash for group_hash in all_group_hashes], 0]
+        )
+
     # We do not want to delete split hashes as they are necessary for keeping groups... split.
-    GroupHash.objects.filter(
-        project_id=project.id, group__id__in=group_ids, state=GroupHash.State.SPLIT
-    ).update(group=None)
-    GroupHash.objects.filter(project_id=project.id, group__id__in=group_ids).exclude(
-        state=GroupHash.State.SPLIT
-    ).delete()
+    all_group_hashes.filter(state=GroupHash.State.SPLIT).update(group=None)
+    all_group_hashes.exclude(state=GroupHash.State.SPLIT).delete()
```

Comment:
I'm not sure if these querysets weren't intentionally not reused for a reason, so i'd keep these the way they were, and define your own queryset within the feature flag.